packages/core/content-releases/admin/src/components/ReleaseModal.tsx

```diff
@@ -99,3 +237,90 @@ export const ReleaseModal = ({
     </ModalLayout>
   );
 };
+
+/**
+ * Generates the list of timezones and user's current timezone(system timezone)
+ */
+interface ITimezoneOption {
+  offset: string;
+  value: string;
+}
+
+const getTimezones = (selectedDate: Date) => {
+  const timezoneList: ITimezoneOption[] = Intl.supportedValuesOf('timeZone').map((timezone) => {
+    // Timezone will be in the format GMT${OFFSET} where offset could be nothing,
+    // a four digit string e.g. +05:00 or -08:00
+    const offsetPart = new Intl.DateTimeFormat('en', {
+      timeZone: timezone,
+      timeZoneName: 'longOffset',
+    })
+      .formatToParts(selectedDate)
+      .find((part) => part.type === 'timeZoneName');
+
+    const offset = offsetPart ? offsetPart.value : '';
+
+    // We want to show time based on UTC, not GMT so we swap that.
+    let utcOffset = offset.replace('GMT', 'UTC');
+
+    // For perfect UTC (UTC+0:00) we only get the string UTC, So we need to append the 0's.
+    if (!utcOffset.includes('+') && !utcOffset.includes('-')) {
+      utcOffset = `${utcOffset}+00:00`;
+    }
+
+    // Offset and timezone are concatenated with '-', so to split and save the required timezone in DB
+    return { offset: utcOffset, value: `${utcOffset}-${timezone}` } satisfies ITimezoneOption;
```

Comment:
I have updated the code so the changes made in date reflects in timezone. 
Though I faced 1 issue to update the combobox value, so I changed value which is `<timezone>`(Europe/Paris) to `<offset-timezone>`(UTC+01:00-Europe/Paris) and `textValue` prop added on combobox.